R/content.R

```diff
@@ -270,6 +270,37 @@ method(contents_markdown, ContentUploaded) <- function(content) {
   sprintf("`[uploaded %s file]`", content@mime_type)
 }
 
+# Thinking ---------------------------------------------------------------------
+
+#' @rdname Content
+#' @export
+ContentThinking <- new_class(
+  "ContentThinking",
+  parent = Content,
+  properties = list(
+    thinking = prop_string(),
+    extra = class_list
+  )
+)
+
+method(format, ContentThinking) <- function(x, ...) {
+  paste("<thinking>\n", x@thinking, "\n</thinking>\n")
+}
+
+method(contents_text, ContentThinking) <- function(content) {
+  content@thinking
+}
+
+method(contents_html, ContentThinking) <- function(content) {
+  check_installed("commonmark")
+  commonmark::markdown_html(content@thinking)
+}
+
+method(contents_markdown, ContentThinking) <- function(content) {
+  content@thinking
```

Comment:
Along the lines of @simonpcouch's comment, I think it's worth having it makes sense to include the `<thinking>` tags in one of the text representations. I'd recommend `contents_markdown()`

```suggestion
  paste("<thinking>\n", x@thinking, "\n</thinking>\n")
```

which would let Simon call `contents_markdown()` on the turn to get the version with the tags.